assignment-2/src/presenter/UsersListPresenter.js

```diff
@@ -1,29 +1,58 @@
 import model from "../model/UserModel";
+import RestClient from "../rest/RestClient";
+import AnswerRestClient from "../rest/AnswerRestClient";
+import WebSocketListener from "../ws/WebSocketlistener";
+import questionModel from "../model/QuestionModel";
+import answerModel from "../model/AnswerModel";
 
 class UsersListPresenter {
     onCreate() {
         model.addUser(model.state.newUser.userName, model.state.newUser.password);
-        model.changeNewUserProperty("userName","");
-        model.changeNewUserProperty("password","");
+        model.changeNewUserProperty("userName", "");
+        model.changeNewUserProperty("password", "");
     }
 
     onChange(property, value) {
         model.changeNewUserProperty(property, value);
     }
     onLogin() {
         //eval('debugger');
-        for(let index = 0; index < model.state.users.length; index++) {
-            if(model.state.users[index].userName === model.state.newUser.userName) {
-                if(model.state.users[index].password === model.state.newUser.password) {
-                    model.changeCurrentUserProperty("userName",model.state.newUser.userName);
-                    window.location.assign("#/questions");
-                }
+        model.state.client = new RestClient(model.state.newUser.userName, model.state.newUser.password);
+        model.state.answerClient = new AnswerRestClient(model.state.newUser.userName, model.state.newUser.password);
+        model.state.listener = new WebSocketListener(model.state.newUser.userName, model.state.newUser.password);
+        model.state.listener.on("event", event => {
```

Comment:
This is a pretty big function -> might have made sense to define it somewhere else (maybe in its own file) and just reference it here.